### ISSUE TITLE
Unable To Use $ref in OAS Parameter Specifications

### ISSUE THREAD

**Issue by ncovercash** [aml-org/amf#1408]
* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

Currently, `$ref` is improperly resolved for relative paths within Parameters

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

> _The extent of AMF being used is, effectively, just `amf.OASConfiguration.OAS30().baseUnitClient().parseDocument("file://foo.yaml")`._

#### _Since this is a schema issue, I created a minimal repo showing the bug:_

Using the [OAS 3.0 petstore example](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore.yaml), the parsing engine breaks when attempting to load local references to parameters.  An error is reported that the reference is unresolved.

I have made https://github.com/ncovercash/amf-ref-demo with a full demo/MVE of this issue, with the `main.js` running `.parseDocument` and `.validate`.

The `combined-petstore` is a working example with only schemas separated from the main specification file.

The `separate-petstore` separates a Path Item from the main file (the /pets) endpoint into `paths/pets.yaml` -- this is valid per [the spec](https://swagger.io/specification/#path-item-object).  However, on this, `.parseDocument` fails, reporting that the references in `pets.yaml` is unresolved.

### **What is the expected behavior?**

The `separate-petstore` example, from above, would parse and validate correctly.

### **What is the motivation/use case for changing the behavior? (for feature requests)**

N/A

### **Please tell us about your environment:**
  
  - AMF Version: 5.0.8
  - AMF Distribution: JS 
  - Operating System: Ubuntu 20.04
  - Link to your project: https://github.com/ncovercash/amf-ref-demo

### **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, Gitter, etc)

This is related to #1338, however, parameter references were out of scope to the original issue.

Here is the full error report:

```
Parsing file:///Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml
parsingResult.conforms: false
parsingResult.results.length: 2
Violation: 'name' property is required in a parameter.
Violation: Cannot find parameter or payload reference ../parameters/limit.yaml
Model: file:///Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml
Conforms: false
Number of results: 2

Level: Violation

- Constraint: http://a.ml/vocabularies/amf/parser#parameter-name-required
  Message: 'name' property is required in a parameter.
  Severity: Violation
  Target: file:///Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml#/web-api/endpoint/%2Fpets/supportedOperation/get/listPets/expects/request/parameter/parameter/default-binding/parameter_1
  Property: 
  Range: [(7,6)-(8,0)]
  Location: file:///Users/ncovercash/amf-ref-demo/separate-petstore/paths/pets.yaml

- Constraint: http://a.ml/vocabularies/amf/parser#unresolved-parameter
  Message: Cannot find parameter or payload reference ../parameters/limit.yaml
  Severity: Violation
  Target: file:///Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml#/web-api/endpoint/%2Fpets/supportedOperation/get/listPets/expects/request/parameter/parameter/default-binding/parameter_1
  Property: 
  Range: [(7,6)-(8,0)]
  Location: file:///Users/ncovercash/amf-ref-demo/separate-petstore/paths/pets.yaml

--------
Validating
validationResult.conforms: false
validationResult.results.length: 1
Violation: 'in' property of a Parameter object is mandatory
ModelId: file:///Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml
Profile: OAS 3.0
Conforms: false
Number of results: 1

Level: Violation

- Constraint: http://a.ml/vocabularies/amf/parser#Parameter-binding-minCount
  Message: 'in' property of a Parameter object is mandatory
  Severity: Violation
  Target: http://amferror.com/#errorParameter/file://Users/ncovercash/amf-ref-demo/separate-petstore/petstore.yaml#/web-api/endpoint/%2Fpets/supportedOperation/get/listPets/expects/request/parameter/parameter/default-binding/parameter_1/parameter/default-binding/default-name
  Property: http://a.ml/vocabularies/apiContract#binding
  Range: [(7,6)-(8,0)]
  Location: file:///Users/ncovercash/amf-ref-demo/separate-petstore/paths/pets.yaml
```

**Comment by arielmirra** [comment]
Hi @ncovercash! This definitely a bug. I've created a [bug report](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000xNqp9YAC/view) assigned to @hghianni as he solved the PR #1338 you linked. We're doing the AMF release, so this fix will not be available in the next AMF release, but in the next one. I'll keep you updated anyway. Thank you!

**Comment by hghianni** [comment]
Hi @ncovercash! This issue has been fixed and will be present in the next AMF release version 5.0.10, probably June 29. Thank you!
